app/jsx/pages/SearchPage.jsx

```diff
@@ -360,72 +366,91 @@ class FacetFieldset extends React.Component {
     }
   }
 
-  /* Gather all IDS of facets (converts tree to flat hash) */
-  getFacetVals = facets =>{
+  /* Gather all IDS of facets already in checkedHash (converts tree to flat hash) */
+  checkedFacetsFromTree = (facets, checkedHash) =>{
     let hash = {}
     this.traverse(facets, function(node){
-      if (!hash[node.value])
+      if (node.value in checkedHash)
         hash[node.value] = true
     })
     return hash
   }
 
+  /* Checked items will percolate to top */
   sliceArrangeFacets(facets)
   {
-    /* Get first set of 6 facets, undiscerning: some may be checked, some may not */
-    let slicedFromTop = this.facetsSliced(facets, 6)
     if (!this.props.query.filters)
-      return slicedFromTop
-
-    /* Get their values. When we shift/prioritize checked facets to the top of the list,
-       we want to make sure they're not already present in this set of initial sliced */
-    let slicedIds = this.getFacetVals(slicedFromTop)
+      return this.facetsSliced(facets, MAX_SET_SIZE)     /* Just return first set of 6 facets */
 
     /* Gather values of checked facets */
     let checkedHash = {}
     for (let filter of this.props.query.filters)
       checkedHash[filter.value] = true
-    let checked_count = Object.keys(checkedHash).length
 
     let checked = []
-    /* Can't use Generator function here since it's not supported by IE */
+    /* Collect checked facets into variable 'checked' (recursive)
+     * Can't use Generator function here since it's not supported by IE */
```

Comment:
Is that true even when we're using Babel? But in any case, this function is nice and clear, so it's fine.

Reply:
Ah I didn't take that into consideration.